scipy/linalg/tests/test_solvers.py

```diff
@@ -577,6 +577,22 @@ def test_infeasible(self):
         assert_raises(LinAlgError, solve_continuous_are, A, B, Q, R)
 
 
+class TestSolveCommonAre:
+    @pytest.mark.parametrize("solver", [solve_continuous_are, solve_discrete_are])
+    def test_with_skipped_array_argument_gh23336(self, solver):
+        # gh-23336 reported a failure when optional argument `e` was skipped
+        A = np.array([[-0.9, 0.25], [0, -1.1]])
+        B = np.array([[0.23], [0.45]])
+        Q = np.eye(2)
+        R = np.atleast_2d(0.45)
+        E = np.eye(2)
+        S = np.array([[0.1], [0.2]])
+
+        res = solver(A, B, Q, R, s=S)
+        ref = solver(A, B, Q, R, E, S)
+        np.testing.assert_allclose(res, ref)
```

Comment:
This test does indeed fail when the source patch is reversed, and in a manner consistent with the original ticket, with `ValueError: Matrix e should be square.` So, that's a good sign. It is also good that no old test cases were modified, only new cases were added.

So, mechanically it looks pretty good to me. The CI failure is just https://github.com/scipy/scipy/issues/23689. That said, I've not been super active in the batching work apart from complaining about error handling things, so may make sense for someone else to give it a quick look.

Matt--how severe is this? Does it justify, on its own, another bug fix release for example?